index.php

```diff
@@ -0,0 +1,157 @@
+ *
+ *     This plugin is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     any later version.
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( 'Tribe__Extension__Advanced_iCal_Export' )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Tribe__Extension__Advanced_iCal_Export extends Tribe__Extension {
+
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Main' );
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			// Don't forget to generate the 'languages/tribe-ext-advanced-ical-export.pot' file
+			load_plugin_textdomain( 'tribe-ext-advanced-ical-export', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			/**
+			 * Protect against fatals by specifying the required minimum PHP
+			 * version. Make sure to match the readme.txt header.
+			 */
+			$php_required_version = '5.6';
+
+			if ( version_compare( PHP_VERSION, $php_required_version, '<' ) ) {
+				if (
+					is_admin()
+					&& current_user_can( 'activate_plugins' )
+				) {
+					$message = '<p>';
+					$message .= sprintf( __( '%s requires PHP version %s or newer to work. Please contact your website host and inquire about updating PHP.', 'tribe-ext-advanced-ical-export' ), $this->get_name(), $php_required_version );
+					$message .= sprintf( ' <a href="%1$s">%1$s</a>', 'https://wordpress.org/about/requirements/' );
+					$message .= '</p>';
+					tribe_notice( $this->get_name(), $message, 'type=error' );
+				}
+				return;
+			}
+
+			// Insert filters and hooks here
+			add_action( 'pre_get_posts', array( $this, 'filter_ical_query' ) );
+		}
+
+		/**
+		 * Filtering the query for dates
+		 *
+		 * @param $query
+		 */
+		function filter_ical_query( $query ) {
+
+			if ( ! isset( $_GET['ical'] )
+			     || ! isset( $query->tribe_is_event_query )
+			     || ! $query->tribe_is_event_query ) {
+				return;
+			}
+
+			$tribe_display	= $_GET[ 'tribe_display' ];
+			$start_date	    = isset( $_GET[ 'start_date' ] ) ? $_GET[ 'start_date' ] : "";
+			$end_date	    = isset( $_GET[ 'end_date' ] )   ? $_GET[ 'end_date' ]   : "";
+
+			if( $tribe_display === 'custom' ) {
+
+				// Check if there is a start_date set
+				if( isset( $start_date ) && !empty( $start_date ) ) {
```

Comment:
Same as mentioned before: add a space between if and the opening parenthesis here: `if ( isset( $start_date ) && !empty( $start_date ) ) {`